src/index.ts

```diff
@@ -7,24 +7,109 @@ const points = new Array<{ x: number, y: number, segmentLength: number }>();
 let unclosedLength = 0;
 let unclosedPath = new Path2D();
 
-const fftSize = 4096;
-const fft = new FFT(fftSize);
-const input = fft.createComplexArray() as number[];
-const output = fft.createComplexArray() as number[];
+let fftSize = 4096;
+let fft = new FFT(fftSize);
+let input = fft.createComplexArray() as number[];
+let output = fft.createComplexArray() as number[];
 const components = new Array<{ frequency: number, magnitude: number, phase: number }>();
 const lines = new Array<{ x: number, y: number }>();
 let parameter = 0;
 let complexity = 0;
 let circles = false;
 let hasCapture = false;
 
+const parameterSlider = document.getElementById('parameter-slider') as HTMLInputElement;
+parameterSlider.oninput = function() {
+    parameter = parameterSlider.valueAsNumber;
+    redraw();
+};
+const complexityNumber = document.getElementById('complexity-number') as HTMLInputElement;
+complexityNumber.oninput = function() {
+    complexity = complexityNumber.valueAsNumber;
+    redraw();
+};
+const complexityCircles = document.getElementById('complexity-circles-check') as HTMLInputElement;
+complexityCircles.oninput = function() {
+    circles = complexityCircles.checked;
+    redraw();
+};
+
 function updateCanvasSize() {
     canvas.width = window.devicePixelRatio * canvas.clientWidth;
     canvas.height = window.devicePixelRatio * canvas.clientHeight;
 }
 
+function loadLocation() { // Inspiration from https://stackoverflow.com/questions/901115/how-can-i-get-query-string-values-in-javascript/21152762#21152762 (qd's not stored)
+    window.location.search?.substr(1).split('&')
+        .forEach(item => {
+            switch (item) {
+                case 'circles':
+                    complexityCircles.checked = true;
+                    break;
+
+                default:
+                    const [k, v] = item.split('=');
+                    if (v !== null) { // Restriction to valued keys
+                        const w = v && decodeURIComponent(v);
+                        switch (true) {
+                            case (k === 'pt'):
```

Comment:
Just curious, is there an advantage to this over `switch (k)`?

Reply:
Of course: if there is the need for a `k AND w`-based case.
...
Okay, this is likely not gonna happen anytime soon, I'll remove that. :)

Reply:
065c652

Reply:
I would move the AND part inside the case then, if I follow what you're saying.